iree/tools/init_dialects.h

```diff
@@ -24,7 +26,9 @@ namespace iree_compiler {
 inline void registerAllDialects(DialectRegistry &registry) {
   registerMlirDialects(registry);
   registerIreeDialects(registry);
+#ifdef IREE_HAVE_TORCH_MLIR_DIALECTS
```

Comment:
One last issue is that when the `init_dilalects.h` is included from https://github.com/google/iree/blob/main/iree/tools/iree_compile_lib.cc#L17 the macro is not defined. Trying to figureout why.

Reply:
Oh, actually when invoked through `iree-build/iree/tools/iree-compile` this works fine. It's just issue when using `iree-build/compiler-api/python_package/iree/compiler/tools/../_mlir_libs/iree-compile`.

Reply:
Maybe edit the deps in here also? https://github.com/google/iree/blob/2453fe5c2b7ebad42217b3a052029d0efdf5b298/llvm-external-projects/iree-compiler-api/lib/CAPI/CMakeLists.txt#L17-L37